src/Microsoft.OData.Client/EntityTracker.cs

```diff
@@ -576,4 +576,12 @@ private void ValidateDuplicateIdentity(Uri identity, EntityDescriptor descriptor
             }
         }
     }
+
+    public class CustomComparer<TObject> : IEqualityComparer<TObject> where TObject : class

```

Comment:
```suggestion
    public class EntityDescriptorComparer<T> : IEqualityComparer<T> where T: class
```
Since the class is public do add some class documentation for it.